crates/block-producer/src/mempool/state.rs

```diff
@@ -197,6 +199,7 @@ impl InflightState {
 struct AccountUpdates {
     from: HashMap<Word, NodeId>,
     to: HashMap<Word, NodeId>,
+    pass_through: HashMap<Word, HashSet<NodeId>>,
 }
```

Comment:
Not for this PR (and not related to the current changes), but I do find this structure a bit difficult to reason about. My main issue is that the structure does not enforce that all updates are contiguous (assuming this is the invariant we want to enforce). At the very least, we should expand the doc comment here to make things a bit more explicit - i.e., do we expect the updates to be contiguous? If yes, how this is enforced? If no, why is this OK?

But also, I do think that potentially a more liner structure could be a bit easier to understand - e.g., something like:
```rust
struct AccountUpdates {
    init_state: Word,
    updates: Vec<(NodeId, Word)>,
    pass_through: HashMap<Word, HashSet<NodeId>>,
}
```